File: src/main/scala/com/abaddon83/cardsGames/shares/players/PlayerIdentity.scala
package com.abaddon83.cardsGames.shares.players

import java.util.UUID

import com.abaddon83.libs.ddd.AggregateUUIDIdentity

case class PlayerIdentity private (override protected val id: UUID) extends AggregateUUIDIdentity {

}

object PlayerIdentity {
  def apply(): PlayerIdentity = new PlayerIdentity(UUID.randomUUID())

  def apply(uuid: UUID): PlayerIdentity = new PlayerIdentity(uuid)

  def apply(uuidString: String): PlayerIdentity = {
    assert(AggregateUUIDIdentity.isValidUUIDString(uuidString),s"${uuidString} is not a valid UUID")
    new PlayerIdentity(UUID.fromString(uuidString))
  }
}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/models/players/PlayerIdentity.kt
package com.abaddon83.burracoGame.commandModel.models.players

import com.abaddon83.utils.ddd.UUIDIdentity
import java.util.*


data class PlayerIdentity constructor(val id: UUID) : UUIDIdentity(id) {

    companion object Factory {
        fun create(): PlayerIdentity = PlayerIdentity(UUID.randomUUID())
        fun create(uuidString: String): PlayerIdentity {
            val uuid = UUID.fromString(uuidString)
            return PlayerIdentity(uuid)
        }
    }
}